File: src/main/java/com/example/reactive/chapter1/listenable/ShoppingCartService.java
package com.example.reactive.chapter1.listenable;

import com.example.reactive.chapter1.commons.Input;
import com.example.reactive.chapter1.commons.Output;

import org.springframework.util.concurrent.ListenableFuture;

public interface ShoppingCartService {

    ListenableFuture<Output> calculate(Input input);
}

File: src/main/kotlin/wmp/edu/reactive/store/listenable/ShoppingCartService.kt
package wmp.edu.reactive.store.listenable

import org.springframework.util.concurrent.ListenableFuture
import wmp.edu.reactive.store.common.Input
import wmp.edu.reactive.store.common.Output

interface ShoppingCartService {
    fun calculate(input: Input): ListenableFuture<Output>
}
